runtime/gc_glue_java/ArrayletObjectModel.cpp

```diff
@@ -110,3 +115,38 @@ GC_ArrayletObjectModel::getArrayletLayout(J9Class* clazz, UDATA dataSizeInBytes,
 	}
 	return layout;
 }
+
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+bool
+GC_ArrayletObjectModel::freeDoubleMapping(MM_EnvironmentBase *env, J9IndexableObject *objectPtr, struct J9PortVmemIdentifier *identifier)
+{
+	PORT_ACCESS_FROM_ENVIRONMENT(env);
+	void *handle = identifier->handle;
+	void *arrayletLeaveAddrs[J9_GC_ARRAYLET_ALLOC_THRESHOLD];
+	void *addressToFree = NULL;
+	if (NULL == handle) {
+		GC_ArrayletLeafIterator arrayletLeafIterator((J9JavaVM *)_omrVM->_language_vm, objectPtr);
+		GC_SlotObject *slotObject = NULL;
+		uintptr_t count = 0;
+
+		while (NULL != (slotObject = arrayletLeafIterator.nextLeafPointer())) {
+			void *currentLeaf = slotObject->readReferenceFromSlot();
+			/* In some corner cases the last leaf might be NULL therefore we must ignore it */
+			if (NULL == currentLeaf) {
+				break;
+			}
+			count++;
+			arrayletLeaveAddrs[count] = currentLeaf;
+		}
+		arrayletLeaveAddrs[0] = (void *)count;
+		identifier->handle = arrayletLeaveAddrs;
+	} else {
+		addressToFree = handle;
+	}
+	j9vmem_free_memory(identifier->address, identifier->size, identifier);
```

Comment:
hm.... I expected a call to updateDoubleMappedLeaves() sometime before this free()!?

Reply:
nor I see j9vmem_free_memory calling it

Reply:
`j9vmem_free_memory` calls `updateDoubleMappedLeaves()` in the OMR side as `omrvmem_free_memory` right at the end of the method.